python_modules/dagstermill/dagstermill/__init__.py

```diff
@@ -138,32 +113,45 @@ def marshal_value(runtime_type, value, target_file):
         check.failed('Unsupported type {name}'.format(name=runtime_type.name))
 
 
-MANAGER_FOR_NOTEBOOK_INSTANCE = Manager()
-
-
-def declare_as_solid(pipeline_def, solid_def_name):
-    return MANAGER_FOR_NOTEBOOK_INSTANCE.declare_as_solid(pipeline_def, solid_def_name)
+def declare_as_solid(repo_def, solid_def_name):
+    return MANAGER_FOR_NOTEBOOK_INSTANCE.declare_as_solid(repo_def, solid_def_name)
 
 
-def define_context(inputs=None):
-    return MANAGER_FOR_NOTEBOOK_INSTANCE.define_context(inputs)
+def yield_result(value, output_name='result'):
+    return MANAGER_FOR_NOTEBOOK_INSTANCE.yield_result(value, output_name)
 
 
-def get_input(dm_context, input_name):
-    check.param_invariant(isinstance(dm_context, (InMemoryDagstermillContext, str)), 'dm_context')
-    return MANAGER_FOR_NOTEBOOK_INSTANCE.get_input(dm_context, input_name)
+def populate_context(dm_context):
+    dm_context_data = dm_context
+    pipeline_def = MANAGER_FOR_NOTEBOOK_INSTANCE.get_pipeline(dm_context_data['pipeline_name'])
+    check.inst(pipeline_def, PipelineDefinition)
+    solid_def_name = dm_context_data['solid_def_name']
+    check.invariant(pipeline_def.has_solid_def(solid_def_name))
+    solid_def = pipeline_def.solid_def_named(solid_def_name)
+    MANAGER_FOR_NOTEBOOK_INSTANCE.solid_def = solid_def
```

Comment:
generally with these mutative things I like to package assignments up into a single method call so that it is clear that we are effectively transitioning the "manager" into a different state. we actually might want to consider modeling the manager as a state machine more directly.

at minimum let's make all the public properties of the manager readonly and only set the variables in method calls to the manager.